rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -363,6 +364,33 @@ __rmw_destroy_node(
   return result_ret;
 }
 
+rmw_ret_t
+__rmw_node_assert_liveliness(
+  const char * identifier,
+  const rmw_node_t * node)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    node,
+    node->implementation_identifier,
+    identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+
+  auto node_info = static_cast<CustomParticipantInfo *>(node->data);
+  if (nullptr == node_info) {
+    RMW_SET_ERROR_MSG("node info handle is null");
+    return RMW_RET_ERROR;
+  }
+  if (nullptr == node_info->participant) {
+    RMW_SET_ERROR_MSG("node internal participant is invalid");
+    return RMW_RET_ERROR;
+  }
+
+  // node_info->participant->assert_liveliness();
```

Comment:
I guess this is planed for the future? But should this function not fail in the meantime?

Reply:
Sorry, I see that it does fail with `RMW_RET_UNSUPPORTED`, but maybe consider still setting the error msg in this case.

Reply:
Added error message.